client/src/components/viewers/types.ts

```diff
@@ -0,0 +1,18 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { Translatable } from 'megashark-lib';
+
+export interface FileControlsDropdownItemContent {
+  id: number;
+  label: Translatable;
+  icon?: string;
+  children?: FileControlsDropdownGroup;
+  dismissPopover?: boolean;
+  dismissToParent?: boolean;
+  callback?: () => any;
+}
+
+export interface FileControlsDropdownGroup {
+  items: FileControlsDropdownItemContent[];
+  activeItem: number | undefined;
```

Comment:
This tells which item is the active one? Why not add an optional `isActive?: boolean` to the item? It would mean that you could just have `Array<FileControlsDropdownItemContent>` instead of creating a whole `FileControlsDropdownGroup` interface.
It would make the API more intuitive to use, and it would probably makes the code simpler (not necessarily shorter because you'd have to go through the list to toggle the selection, but simpler).